### ISSUE TITLE
 Lesson 1: Image classification

### ISSUE THREAD

**Issue by datalass1** [datalass1/fastai#19]
The key outcome of this lesson is that we'll have trained an image classifier which can recognize pet breeds at state of the art accuracy. The key to this success is the use of transfer learning, which will be a key platform for much of this course. We'll also see how to analyze the model to understand its failure modes. In this case, we'll see that the places where the model is making mistakes is in the same areas that even breeding experts can make mistakes.

**Comment by datalass1** [comment]
### Some info on GPUs
By [Rachel Thomas](https://www.fast.ai/2017/11/16/what-you-need/)
CUDA and OpenCL are the two main ways for programming GPUs. CUDA is by far the most developed, has the most extensive ecosystem, and is the most robustly supported by deep learning libraries. CUDA is a proprietary language created by Nvidia, so it can’t be used by GPUs from other companies. When fast.ai recommends Nvidia GPUs, it is not out of any special affinity or loyalty to Nvidia on our part, but that this is by far the best option for deep learning.

### Introductions 0-10mins
why fastai etc

### Lesson-1 Pets
Always start a jupyter notebook with these three lines of code:
```
%reload_ext autoreload
%autoreload 2
%matplotlib inline
```
the % is special directives to jupyter notebooks. If someone changes the code while running, reload. If someone plots something, plot it in Jupyter notebook.

**PyTorch** is one of the most popular libraries in the world. More modern than Tensorflow. 

**Fastai library** sits on top of PyTorch to do things faster and easily. 
4 Core uses: Computer Vision, natural language text, tabular data and collaborative filtering.
```import *``` is great, does not agree with PEP8 software engineering practice but in Data Science the rules are different, there is a need to interactively experiment.

**Comment by datalass1** [comment]
stopped at 1hr

**Comment by datalass1** [comment]
### Transferring files from Linux instances to local repo
Transferring files using the gcloud command-line tool
gcloud compute scp --recurse /home/ubuntu/git/fastai/dl1/ my-fastai-instance:/home/jupyter/tutorials/fastai/course-v3/nbs/dl1/lesson1v3-pets-20190131.ipynb

ERROR: (gcloud.compute.ssh) Could not fetch resource: - Insufficient Permission
The solution:
It looks like the attempt to SSH is recognising the instance in your project, but the user doesn't have the required permissions to access the machine. Run:
```$ gcloud auth login```

**Comment by datalass1** [comment]
ERROR: (gcloud.compute.scp) Could not fetch resource:
 - The resource 'projects/rek-fastai/zones/us-west1-b/instances/my-fastai-instance' was not found

Tried on local and VM:
```gcloud config list compute/region``` to view the region
```gcloud config set compute/region us-west2``` to set a new region

To see the full GCP config for a project:
```gcloud config list```
To change zone:
```gcloud config set compute/zone us-west2-b```

Finally:
```gcloud compute scp jupyter@my-fastai-instance:~/tutorials/fastai/course-v3/nbs/dl1/lesson1v3-pets-20190131.ipynb ~/home/ubuntu/git/fastai/dl1/```
